frontend/src/components/account/showShareInformationsModal.tsx

```diff
@@ -0,0 +1,92 @@
+import { Text, Divider, Progress, Stack, Group, Flex } from "@mantine/core";
+import { ModalsContextProps } from "@mantine/modals/lib/context";
+import { MyShare } from "../../types/share.type";
+import moment from "moment";
+import { byteToHumanSizeString } from "../../utils/fileSize.util";
+import CopyTextField from "../upload/CopyTextField";
+import { FileMetaData } from "../../types/File.type";
+
+const showShareInformationsModal = (
+  modals: ModalsContextProps,
+  share: MyShare,
+  appUrl: string,
+  maxShareSize: number | string
+) => {
+  const link = `${appUrl}/share/${share.id}`;
+
+  let shareSize: number = 0;
+  for (let file of share.files as FileMetaData[])
+    shareSize += parseInt(file.size);
+
+  if (typeof maxShareSize === "string") maxShareSize = parseInt(maxShareSize);
+
+  const formattedShareSize = byteToHumanSizeString(shareSize);
+  const formattedMaxShareSize = byteToHumanSizeString(maxShareSize);
+  const shareProgress = (shareSize / maxShareSize) * 100;
```

Comment:
I would rename that because `sharePrgoress` could lead to confusion
```suggestion
  const shareSizeProgress = (shareSize / maxShareSize) * 100;
```